### PR TITLE
Solve the problem of address delivery in service discovery scenarios

### PR DESCRIPTION
**Pull request by xiaoheng1**
## What is the purpose of the change

Solve the problem of address delivery in service discovery scenarios

see #8491 

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by guohao — CHANGES_REQUESTED** [review]
Root cause:
ClusterInvoker may call 
`org.apache.dubbo.rpc.cluster.support.AbstractClusterInvoker#invokeWithContext` in another thread, but `ConsumerUrl` was not copied to new context.

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/InstanceAddressURL.java [hunk 1]
```diff
@@ -62,30 +62,38 @@ public MetadataInfo getMetadataInfo() {
...
-        return RpcContext.getServiceContext().getInterfaceName();
+        String serviceInterface = RpcContext.getServiceContext().getInterfaceName();
+        return StringUtils.isEmpty(serviceInterface) ? super.getParameter(INTERFACE_KEY) : serviceInterface;
     }
 
+    @Override
     public String getGroup() {
-        return RpcContext.getServiceContext().getGroup();
+        String group = RpcContext.getServiceContext().getGroup();
+        return StringUtils.isEmpty(group) ? super.getGroup() : group;
```
[the remark this answers is not in the document]
Comment (xiaoheng1) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/InstanceAddressURL.java]: Okay, I set consumerUrl to ServiceContext in AbstractClusterInvoker.

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/AbstractClusterInvoker.java [hunk 2]
```diff
@@ -333,10 +351,15 @@ protected LoadBalance initLoadBalance(List<Invoker<T>> invokers, Invocation invo
...
+        setContext(invoker, null);
+    }
+
+    private void setContext(Invoker<T> invoker, URL consumerUrl) {
         RpcContext context = RpcContext.getServiceContext();
         context.setInvoker(invoker)
-                .setRemoteAddress(invoker.getUrl().getHost(), invoker.getUrl().getPort())
-                .setRemoteApplicationName(invoker.getUrl().getRemoteApplication());
+            .setRemoteAddress(invoker.getUrl().getHost(), invoker.getUrl().getPort())
+            .setRemoteApplicationName(invoker.getUrl().getRemoteApplication())
```
Comment (AlbumenJ) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/AbstractClusterInvoker.java:361]: needless to set remote-address and remote-application-name

**Review by guohao — APPROVED** [review]

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/AbstractClusterInvoker.java (hunk 2 above)
Comment (xiaoheng1) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/AbstractClusterInvoker.java:361]: ok，i will remove it.

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Pull request merged**